moshi/src/main/java/com/squareup/moshi/Moshi.java

```diff
@@ -227,6 +228,26 @@ public Builder add(Object adapter) {
       return add(AdapterMethodsFactory.get(adapter));
     }
 
+    /**
+     * @param fallback the default value to use when a string is deserialized
+     * and does not match an existing enum value.
+     */
+    public <T extends Enum<T>> Builder addEnumFallback(final T fallback) {
+      if (fallback == null) {
+        throw new NullPointerException("fallback == null");
+      }
+      final Class<T> enumType = fallback.getDeclaringClass();
```

Comment:
or could accept the class as a parameter and allow null values. not sure. i tend to use non-null enum types and have another enum value for the sentinel value of unknown.